### PR TITLE
Implement dashboard name

### PR DESCRIPTION
**Pull request by bjarnef**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
This PR add the option to set a dashboard name without need to add localization.
The `Name` property act as fallback, so it won't affect current implementation of `IDashboard` where it still use translation if prevent and fallback to `Name` prooperty, but if not implemented, null or empty value is shown the default localized value, which is `[alias]`.

It also makes it a bit more consistent with other concepts in core, e.g. ContentApp where a name can be set without translation:
https://github.com/umbraco/Umbraco-CMS/blob/c6d01178c06a519423b3c37fea7bbdd0e536319f/src/Umbraco.Core/Models/ContentEditing/ContentApp.cs#L18

### Test notes

Extract following folder to `Umbraco.Web.UI` project:
[Dashboards.zip](https://github.com/umbraco/Umbraco-CMS/files/13527278/Dashboards.zip)

Furthermore extract this folder to `App_Plugins`:
[Our.Umbraco.Dashboards.zip](https://github.com/umbraco/Umbraco-CMS/files/13527388/Our.Umbraco.Dashboards.zip)

With no localization of `dashboardTabs_{alias}` and `Name` property set on implementation of `IDashboard`:

image

When comment out `Name` property in `OutdatedContentDashboard` (like if developers haven't implemented it) and no translation of `dashboardTabs_{alias}`:

image

With translation and with or without implementation of `Name` property:

image

### CODE REVIEW

File: src/Umbraco.Core/Dashboards/IDashboardSlim.cs [hunk 1]
```diff
@@ -13,6 +13,12 @@ public interface IDashboardSlim
     [DataMember(Name = "alias")]
     string? Alias { get; }
 
+    /// <summary>
+    ///     Gets the name of the dashboard.
+    /// </summary>
+    [DataMember(Name = "name")]
+    string? Name => null;
```
Comment (bjarnef) [inline src/Umbraco.Core/Dashboards/IDashboardSlim.cs:20]: Do we want `null` or empty string here?

**Comment by nul800sebastiaan** [comment]
Hi @bjarnef, as we talked about a few weeks ago, I'll close this one while we focus on getting Bellissima in good shape.

I'll add the generic comment for people who are reading along with this:

As you may or may not know, we're working on a rewrite of the Umbraco backoffice, codenamed Bellissima (https://umbraco.com/blog/bellissima-preview-releases-of-the-new-backoffice/) and we're trying to make sure that it has feature parity with the current AngularJS-based backoffice.

We're at a point in the development phase where we want to start wrapping up a beta version to ship some time "soon" and we're going to have to implement a feature freeze on the current backoffice so that we don't keep increasing the scope of Bellissima. Unfortunately that means we're going to have to stop accepting PRs like this one that create significant changes to the current backoffice.

As a general rule from now on, we're still happy for features in C# code but anything that gets introduced or refactored in AngularJS probably won't make it. Bugfixes are still great of course!

Thanks for the efforts here and we hope to see you back as a contributor soon! 👍

**Comment by bjarnef** [comment]
@nul800sebastiaan actually the changes here is only in C# although I did some formatting in the controller and view :)

**Comment by nul800sebastiaan** [comment]
@bjarnef Sure, dashboard in new backoffice use a whole different pattern though and dashboard naming works just fine as it is right now, albeit a bit more complicated then we would like. Much better in Bellissima, so we'll focus on that instead of introducting code that is almost immediately obsolete. :)

**Pull request closed without merging**